ledger/test-common/src/main/daml/model/Foo.daml

```diff
@@ -29,3 +32,52 @@ template Foo3
   where
     signatory signatory
     observer observers
+
+
+data ContractDefinition = ContractDefinition
+  with
+    templateId : Int
+    observerUsed : [Bool]
+    archive: Bool
+    payload : Text
+  deriving (Eq, Show)
+
+-- Given a list of boolean flags and another list of equal length,
+-- returns those elements from the second list where the flag is True at the same index
+takeIf : [Bool] -> [a] -> [a]
+takeIf (True :: as) (b :: bs) = b :: takeIf as bs
+takeIf (False :: as) (b :: bs) = takeIf as bs
+takeIf [] [] = []
+takeIf _ _ = error("Unbalanced input")
+
+createContract : Party -> [Party] -> ContractDefinition -> Update ()
+createContract signatory observers definition =
+  do
+    case definition.templateId of
+      1 -> do
+        cid <- create Foo1 with signatory=signatory, observers=(takeIf definition.observerUsed observers), payload=definition.payload
```

Comment:
I think create+archive done in DAML code will result in a transient contract, but I would live to be proven wrong.

Reply:
I thought you would create batches on the proto level:
https://github.com/digital-asset/daml/blob/da76e2825b4b1c78766d8db0a4a68359e95c681e/ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto#L58

Reply:
Then it would be possible to add a CreateAndExerciseCommand that likely would not result in a transient contract.

Reply:
What is a transient contract? When the archive flag is set, this Daml code creates 1 create and 1 archive event in the index database (just verified on Postgres).

I'll try to rewrite the code using batching at proto level, that sounds much simpler than what I did here.

Reply:
I have pushed a change that rewrites the code to use batching at proto level. This is slightly slower on my machine, but much easier to maintain.

Reply:
> What is a transient contract? When the archive flag is set, this Daml code creates 1 create and 1 archive event in the index database (just verified on Postgres).

If you were able to see it in the db, it is not a transient contract after all. My worry was unsubstantiated.